### PR TITLE
Add Drag&Drop (sorting items) in bookmarks

### PR DESCRIPTION
**Pull request by slprime**
nei-bookmarks-sorting

Sorting items in bookmarks.  
For use it need hold shift

issue: https://github.com/GTNewHorizons/NotEnoughItems/issues/16

### CODE REVIEW

File: src/codechicken/nei/BookmarkPanel.java [hunk 1]
```diff
@@ -171,4 +173,68 @@ public int getWidth(GuiContainer gui) {
...
+                }
+
+            } else if (mouseOverSlot != null && mouseOverSlot.slotIndex != sortedStackIndex) {
+
+                int maxStackIndex = Math.max(mouseOverSlot.slotIndex, sortedStackIndex);
+                int slotIndex = mouseOverSlot.slotIndex;
+                ItemStack cacheItem = _items.get(sortedStackIndex);
+                String recipeId = _recipes.get(sortedStackIndex);
+
+                if (sortedStackIndex == maxStackIndex) {
```
Comment (mitchej123) [inline src/codechicken/nei/BookmarkPanel.java:201]: Wouldn't a `_items.add(slotIndex, _items.remove(sortedStackIndex))` handle all of the shifting for you?

Comment (slprime) [reply src/codechicken/nei/BookmarkPanel.java]: thanks, I didn't know that this could be done

**Comment by slprime** [comment]
This pull request includes improvements for https://github.com/GTNewHorizons/NotEnoughItems/pull/173 

Need remove __recipe variable if need run this

**Comment by slprime** [comment]
New PR: https://github.com/GTNewHorizons/NotEnoughItems/pull/174

**Pull request closed without merging**